File: web-app/tests/countrySettingRepository.test.ts
/* eslint-env browser */
import { describe, it, expect, beforeEach } from 'vitest';
import { CountrySettingRepository, type CountrySetting } from '../src/repositories/CountrySettingRepository';

describe('CountrySettingRepository', () => {
  beforeEach(() => {
    localStorage.clear();
  });

  it('saves and loads setting', async () => {
    const repo = new CountrySettingRepository();
    const setting: CountrySetting = {
      iso2: 'US',
      lastCurrency: 'USD',
      acquired: '2024-01-01T00:00:00Z',
      method: 'GPS'
    };
    await repo.save(setting);
    const loaded = await repo.load();
    expect(loaded).toEqual(setting);
  });

  it('returns null when empty', async () => {
    const repo = new CountrySettingRepository();
    const loaded = await repo.load();
    expect(loaded).toBeNull();
  });
});

File: mobile-app/test/country_setting_repository_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:shared_preferences/shared_preferences.dart';
import 'package:mobile_app/models/country_setting.dart';
import 'package:mobile_app/repositories/country_setting_repository.dart';

void main() {
  TestWidgetsFlutterBinding.ensureInitialized();

  group('CountrySettingRepository', () {
    setUp(() {
      SharedPreferences.setMockInitialValues({});
    });

    test('saves and loads setting', () async {
      final repo = CountrySettingRepository();
      final setting = CountrySetting(
        iso2: 'US',
        lastCurrency: 'USD',
        acquired: DateTime.utc(2024, 1, 1),
        method: 'GPS',
      );
      await repo.save(setting);
      final loaded = await repo.load();
      expect(loaded?.iso2, 'US');
      expect(loaded?.lastCurrency, 'USD');
    });

    test('load returns null when empty', () async {
      final repo = CountrySettingRepository();
      final loaded = await repo.load();
      expect(loaded, isNull);
    });
  });
}
